cli/src/run.rs

```diff
@@ -1,10 +1,36 @@
+use std::fs::File;
+
 use crate::CliResult;
 use crate::{Model, Parameters};
 use ansi_term::Color::*;
+use ndarray_npy::{NpzWriter};
 use tract_hir::internal::*;
 #[cfg(feature = "pulse")]
 use tract_pulse::internal::*;
 
+/// Add a tensor entry into a npz file.
+fn npz_add_tensor(npz: &mut NpzWriter<File>, name: String, tensor: &Arc<Tensor>) -> CliResult<()> {
+    match tensor.datum_type() {
+        DatumType::Bool => npz.add_array(name, &tensor.to_array_view::<bool>()?)?,
+        DatumType::U8 => npz.add_array(name, &tensor.to_array_view::<u8>()?)?,
+        DatumType::U16 => npz.add_array(name, &tensor.to_array_view::<u16>()?)?,
+        DatumType::U32 => npz.add_array(name, &tensor.to_array_view::<u32>()?)?,
+        DatumType::U64 => npz.add_array(name, &tensor.to_array_view::<u64>()?)?,
+        DatumType::I8 => npz.add_array(name, &tensor.to_array_view::<i8>()?)?,
+        DatumType::I16 => npz.add_array(name, &tensor.to_array_view::<i16>()?)?,
+        DatumType::I32 => npz.add_array(name, &tensor.to_array_view::<i32>()?)?,
+        DatumType::I64 => npz.add_array(name, &tensor.to_array_view::<i64>()?)?,
+        DatumType::F32 => npz.add_array(name, &tensor.to_array_view::<f32>()?)?,
+        DatumType::F64 => npz.add_array(name, &tensor.to_array_view::<f64>()?)?,
+        DatumType::QI8(_) => npz.add_array(name, &tensor.to_array_view::<i8>()?)?,
```

Comment:
I admit I haven't tested, but i was fairly certain that we don't need it because `to_array_view` calls `check_for_access`, which compare the datum types unquantized:
https://github.com/sonos/tract/blob/main/data/src/tensor.rs#L706 and
`DatumType::QI8(_).unquantized() == DatumType::I8.unquantized()`

I will check to make sure this is working as i thought

Reply:
Ho, ok. Good catch then. Trying to remember if we have something covering this in CI but I don't think so...

Reply:
Hello, sorry for the delay.

I have tested, and removing the unsafe blocks works. I tested it using an identity ONNX model (using other things than ONNX would probably have been better) and applying PR #722 to get quantized cli inputs.

If I were to add this to the CI, how would I go about it?

Reply:
You can add it to onnx/test_cases. It is a bit borderline, as we are actually testing a different crate (cli, not onnx) but we'll live with it.